### PR TITLE
BGL: Document graph_has_property

### LINKED ISSUE #1837: Document graph_has_property
.. and move it outside boost namespace?

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
Move graph_has_property from namespace boost to namespace CGAL and provide some doc for it.

## Release Management

* Affected package(s):
* Issue(s) solved (if any): fix #1837
* Feature/Small Feature (if any):[ here](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Graph_has_property)
* pre-approved by Sloriot 24 January 2018

### CODE REVIEW

File: BGL/include/CGAL/boost/graph/properties.h [hunk 1]
```diff
@@ -31,7 +31,29 @@
 #include <CGAL/basic.h>
 #include <string>
 
- 
+namespace CGAL{
+/// \ingroup PkgBGLProperties
+/// \brief graph_has_property is used to indicate if
+/// a model of `HalfedgesGraph` or `FaceGraph`
```
Comment (MaelRL) [inline BGL/include/CGAL/boost/graph/properties.h]: `HalfedgeGraph` (no `s`).

Comment (maxGimeno) [reply BGL/include/CGAL/boost/graph/properties.h]: oops

**Comment by lrineau** [comment]
That is something that was discussed for a year... but it seems a bit too late to get it inside 4.12. Do you agree this PR targets 4.13?

File: BGL/include/CGAL/boost/graph/properties.h [hunk 2]
```diff
@@ -31,7 +31,29 @@
...
+namespace CGAL{
+/// \ingroup PkgBGLProperties
+/// \brief graph_has_property is used to indicate if
+/// a model of `HalfedgeGraph` or `FaceGraph`
+/// has an internal property associated with the
+/// given `PropertyTag`.
+///
+/// It inherits from `CGAL::Tag_true` if there is a
+/// default internal property map for the
+/// corresponding property tag or from
```
Comment (sloriot) [inline BGL/include/CGAL/boost/graph/properties.h]: I think it should be _and from XXX otherwise_

**Comment by sloriot** [comment]
@lrineau up to you, the PR looks good to me.

**Comment by lrineau** [comment]
OK, let's try to have a quick integration. The approval of the small feature will have to be pretty fast, to avoid delay the release of CGAL-4.12-beta1.

File: BGL/include/CGAL/boost/graph/properties.h [hunk 3]
```diff
@@ -40,7 +40,7 @@ namespace CGAL{
 ///
 /// It inherits from `CGAL::Tag_true` if there is a
 /// default internal property map for the
-/// corresponding property tag or from
+/// corresponding property tag and from
 /// CGAL::Tag_false otherwise.
```
Comment (lrineau) [inline BGL/include/CGAL/boost/graph/properties.h]: You should quote `CGAL::Tag_false`.

**Pull request merged**